src/components/collapsible/_collapsible.scss

```diff
@@ -53,6 +53,7 @@ $collapsible-caret-width: 1.5rem;
     &__title {
       transform: translateY(-1px);
       margin-bottom: 0;
+      font-size: 1rem;
```

Comment:
Need the bold font weight too when the heading tag isn't used
https://deploy-preview-1246--ons-design-system.netlify.app/components/collapsible/examples/collapsible-with-save/